indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java

```diff
@@ -324,12 +324,9 @@ protected void reduce(
             HyperLogLogCollector.makeCollector(ByteBuffer.wrap(value.getBytes(), 0, value.getLength()))
         );
       }
-      Optional<Interval> intervalOptional = config.getGranularitySpec().bucketInterval(new DateTime(key.get()));
 
-      if (!intervalOptional.isPresent()) {
-        throw new ISE("WTF?! No bucket found for timestamp: %s", key.get());
-      }
-      Interval interval = intervalOptional.get();
+      Interval interval = config.getGranularitySpec().getSegmentGranularity().bucket(new DateTime(key.get()));
```

Comment:
You mean ```config.getGranularitySpec().bucketInterval()``` and ```config.getGranularitySpec().getSegmentGranularity().bucket()``` are interchangeable? I'm not sure about that because ```config.getGranularitySpec().bucketInterval()``` finds a proper interval among the given bucket intervals in the granularity spec while ```config.getGranularitySpec().getSegmentGranularity().bucket()``` computes an interval from the input dateTime.

I think ```determinIntervals``` means that we should compute intervals from inputs because bucket intervals is absent in the granularity spec if it's true. Otherwise, we can find proper intervals from the given bucket intervals.